api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -350,4 +355,129 @@ private OrderSet orderSetBuilder(boolean orderSetRetired, boolean orderSetMember
+		assertEquals(initialOrderSetAttributeTypesCount + 0,
+				Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+		Context.getOrderSetService().purgeOrderSetAttributeType(Context.getOrderSetService().getOrderSetAttributeType(2));
+		assertEquals(1, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#retireOrderSetAttributeType(OrderSetAttributeType,String)
+	 */
+	@Test
+	public void retireOrderSetAttributeType_shouldRetireAOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetAttributeType vat = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertFalse(vat.getRetired());
+		assertNull(vat.getRetiredBy());
+		assertNull(vat.getDateRetired());
+		assertNull(vat.getRetireReason());
+		Context.getOrderSetService().retireOrderSetAttributeType(vat, "for testing");
+		vat = Context.getOrderSetService().getOrderSetAttributeType(1);
+		assertTrue(vat.getRetired());
+		assertNotNull(vat.getRetiredBy());
+		assertNotNull(vat.getDateRetired());
+		assertEquals("for testing", vat.getRetireReason());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldCreateANewOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		int initialOrderSetAttributeTypesCount = Context.getOrderSetService().getAllOrderSetAttributeTypes().size();
+		assertEquals(initialOrderSetAttributeTypesCount + 0,
+				Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+		OrderSetAttributeType oat = new OrderSetAttributeType();
+		oat.setName("Another one");
+		oat.setDatatypeClassname(FreeTextDatatype.class.getName());
+		Context.getOrderSetService().saveOrderSetAttributeType(oat);
+		assertNotNull(oat.getId());
+		assertEquals(3, Context.getOrderSetService().getAllOrderSetAttributeTypes().size());
+	}
+
+	/**
+	 * @see OrderSetService#saveOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void saveOrderSetAttributeType_shouldEditAnExistingOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetService service = Context.getOrderSetService();
+		assertEquals(2, service.getAllOrderSetAttributeTypes().size());
+		OrderSetAttributeType oat = service.getOrderSetAttributeType(1);
+		oat.setName("A new name");
+		service.saveOrderSetAttributeType(oat);
+		assertEquals(2, service.getAllOrderSetAttributeTypes().size());
+		assertEquals("A new name", service.getOrderSetAttributeType(1).getName());
+	}
+
+	/**
+	 * @see OrderSetService#unretireOrderSetAttributeType(OrderSetAttributeType)
+	 */
+	@Test
+	public void unretireOrderSetAttributeType_shouldUnretireARetiredOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+		OrderSetService service = Context.getOrderSetService();
+		OrderSetAttributeType oat = service.getOrderSetAttributeType(2);
```

Comment:
What is oat?

Reply:
it was a abbreviation for `orderSetAttributeType` but i have fixed it